File: samples/Lilly/tutorials/broadcaster/flow/outflow/bcast-outflow.c
#include <stdio.h>
#include <unistd.h>
#include <time.h>

int main(int argc, char *argv[]){
	time_t rawtime;
	struct tm *t;

	while(1){
		time(&rawtime);
		t = localtime(&rawtime);		
		myClock = {
			.year: t->tm_year+1990, 
			.month: t->tm_mon,
			.date: t->tm_mday,
			.hour: t->tm_hour,
			.minute: t->tm_min,
			.second: t->tm_sec
		};
		sleep(1);
		//printf("%d\n", myClock.second);
	}

	return 0;
}

File: samples/Lilly/tutorials/broadcaster/flow/outflow/bcast-outflow.js
jdata{
	struct currentTime{
		int year;
		int month;
		int date;
		int hour;
		int minute;
		int second;
	} myClock as logger;

	f as flow with flowFunc of myClock;

	outF as outflow of f;
}

/* FLOW ACTIONS */
(function poll(){
	if(myClock.size()<1){
		console.log("Waiting for a C-Node");
		setTimeout(poll, 1000);
	}
	f.startPush();
})();

function flowFunc(inputFlow){
	return inputFlow;
}

f.setTerminalFunction(function(entry){
	var data = entry.data;
	console.log("Sending data......")
	prettyPrint(data);
});

function prettyPrint(time){
	var output = String(time.year+"-"+time.month+"-"+time.date+" "+time.hour+":"+time.minute+":"+time.second);
	console.log(output);
	return output;
}